chromium_src/chrome/browser/ui/views/tabs/tab_strip.cc

```diff
@@ -24,40 +27,66 @@
+    }                                                                          \
+    if (distance < min_distance) {                                             \
+      min_distance = distance;                                                 \
+      min_distance_index = candidate_index;                                    \
+    }                                                                          \
+    continue;                                                                  \
   }
 
 #define BRAVE_CALCULATE_BOUNDS_FOR_DRAGGED_VIEWS                           \
   if (tabs::features::ShouldShowVerticalTabs(tab_strip_->GetBrowser())) {  \
     std::vector<gfx::Rect> bounds;                                         \
+    int x = 0;                                                             \
     int y = 0;                                                             \
     for (const TabSlotView* view : views) {                                \
-      int x = 0;                                                           \
-      if (view->GetTabSlotViewType() == TabSlotView::ViewType::kTab &&     \
-          view->group().has_value()) {                                     \
-        x = BraveTabGroupHeader::GetLeftPaddingForVerticalTabs();          \
-      }                                                                    \
       const int height = view->height();                                   \
+      if (view->GetTabSlotViewType() == TabSlotView::ViewType::kTab) {     \
+        if (const Tab* tab = static_cast<const Tab*>(view);                \
+            tab->data().pinned) {                                          \
+          /* In case it's a pinned tab, lay out them horizontally */       \
+          bounds.emplace_back(x, y, tabs::kVerticalTabMinWidth, height);   \
+          constexpr int kStackedOffset = 4;                                \
+          x += kStackedOffset;                                             \
+          continue;                                                        \
+        }                                                                  \
+        if (view->group().has_value()) {                                   \
+          /* In case it's a tab in a group, set left padding */            \
+          x = BraveTabGroupHeader::GetLeftPaddingForVerticalTabs();        \
+        }                                                                  \
+      }                                                                    \
       bounds.emplace_back(x, y, TabStyle::GetStandardWidth() - x, height); \
+      /* unpinned dragged tabs are laid out vertically */                  \
       y += height;                                                         \
     }                                                                      \
     return bounds;                                                         \
   }
```

Comment:
I wonder if it's a good time to move bodies from `BRAVE_CALCULATE_INSERTION_INDEX, BRAVE_CALCULATE_BOUNDS_FOR_DRAGGED_VIEWS` into separate functions instead of growing the defines.

Reply:
Yeah, that's a good call. Done.